lib/_stream_readable.js

```diff
@@ -498,7 +495,7 @@ Readable.prototype.read = function(n) {
     ret = null;
 
   if (ret === null) {
-    state.needReadable = state.length <= state.highWaterMark;
+    state.needReadable = !state.ended && state.length <= state.highWaterMark;
```

Comment:
I'm not quite sure if this change is actually needed. But it is needed to keep the logic the same.